### PR TITLE
[zos_copy] Files corrupted after second copy

### LINKED ISSUE #737: [Bug] [zos_copy] File wrongly modified after second zos_copy
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.1

### IBM Enterprise Python

v3.8.x

### IBM z/OS Ansible core Version

v1.5.0

### ansible-version

v2.11.x

### z/OS version

v2.4

### Ansible module

zos_copy

### Bug description

Problem with zos_copy module when I copy a file from a uss directory to another uss directory when the file already exists in dest (Test also with version 1.4.0 and 1.5.0-beta.1 of Ansible z/OS core collection)
Here's my task:
   - name: "Test zos_copy"
     zos_copy:
       src: "/tmp/input/test2.txt"
       dest: "/tmp/wd_uss/txt/test2.txt"
       remote_src: true
       mode: '0755'
       backup: false
       force: true
The content of the file to copy is :
Ceci n est que du text
After first copy, test2.txt content in tmp/wd_uss/txt is ok :
Ceci n est que du text
After second copy (/tmp/wd_uss/txt/test2.txt already exists), file is modified (content unreadable).
!! I don't touch the source, it's the same on the 1st and 2nd launch (/tmp/input/test2.txt)

For information  :
ls -T /tmp/input/test2.txt ==>
t IBM-1047    T=on  /tmp/input/test2.txt
ls -T /tmp/wd_uss/txt/test2.txt ==>
t IBM-1047    T=on  /tmp/wd_uss/txt/test2.txt
When I run the task for the second time, the module returns "changed": true

### Playbook verbosity output.

TASK [ibm.ibm_zos_wazi_deploy.zos_deploy : Test uss_copy] **************************************************************
task path: /mnt/c/Users/058275706/git/wazi-deploy-samples/translator/ansible/ansible_collections/ibm/ibm_zos_wazi_deploy/roles/zos_deploy/tasks/ibm_zos_collections/test.yml:16
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: vcorfma
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="vcorfma"' -o ConnectTimeout=10 -o 'ControlPath="/home/vcorfmat/.ansible/cp/70a06198bc"' 9.30.222.103 '/bin/sh -c '"'"'echo ~vcorfma && sleep 0'"'"''
<9.30.222.103> (0, b'/u/vcorfma\n', b'')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: vcorfma
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="vcorfma"' -o ConnectTimeout=10 -o 'ControlPath="/home/vcorfmat/.ansible/cp/70a06198bc"' 9.30.222.103 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /u/vcorfma/.ansible/tmp `"&& mkdir "` echo /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442 `" && echo ansible-tmp-1681474911.0594854-304-278803071325442="` echo /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442 `" ) && sleep 0'"'"''
<9.30.222.103> (0, b'ansible-tmp-1681474911.0594854-304-278803071325442=/u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442\n', b'')
Using module file /home/vcorfmat/.ansible/collections/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_copy.py
<9.30.222.103> PUT /home/vcorfmat/.ansible/tmp/ansible-local-182tsmuxyly/tmpncu3q_94 TO /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442/AnsiballZ_zos_copy.py
<9.30.222.103> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="vcorfma"' -o ConnectTimeout=10 -o 'ControlPath="/home/vcorfmat/.ansible/cp/70a06198bc"' '[9.30.222.103]'
<9.30.222.103> (0, b'sftp> put /home/vcorfmat/.ansible/tmp/ansible-local-182tsmuxyly/tmpncu3q_94 /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442/AnsiballZ_zos_copy.py\n', b'')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: vcorfma
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="vcorfma"' -o ConnectTimeout=10 -o 'ControlPath="/home/vcorfmat/.ansible/cp/70a06198bc"' 9.30.222.103 '/bin/sh -c '"'"'chmod u+x /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442/ /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442/AnsiballZ_zos_copy.py && sleep 0'"'"''
<9.30.222.103> (0, b'', b'')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: vcorfma
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="vcorfma"' -o ConnectTimeout=10 -o 'ControlPath="/home/vcorfmat/.ansible/cp/70a06198bc"' -tt 9.30.222.103 '/bin/sh -c '"'"'_BPXK_AUTOCVT=ON ZOAU_HOME=/var/usr/lpp/IBM/zoautil LIBPATH=/var/usr/lpp/IBM/zoautil/lib:/var/usr/lpp/IBM/cyp/v3r10/pyz/lib:/lib:/usr/lib:. PATH=/var/usr/lpp/IBM/zoautil/bin:/var/usr/lpp/IBM/cyp/v3r10/pyz/bin:/bin:/var/bin _CEE_RUNOPTS='"'"'"'"'"'"'"'"'FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)'"'"'"'"'"'"'"'"' _TAG_REDIR_ERR=txt _TAG_REDIR_IN=txt _TAG_REDIR_OUT=txt LANG=C /var/usr/lpp/IBM/cyp/v3r10/pyz/bin/python3 /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442/AnsiballZ_zos_copy.py && sleep 0'"'"''
<9.30.222.103> (0, b'\r\n{"size": 23, "changed": false, "src": "/SYSTEM/tmp/input/test2.txt", "dest": "/tmp/wd_uss/txt/test2.txt", "ds_type": "USS", "dest_exists": true, "backup_name": null, "uid": 10149, "gid": 100, "owner": "VCORFMA", "group": "OMVS", "mode": "0755", "state": "file", "invocation": {"module_args": {"src": "/tmp/input/test2.txt", "dest": "/tmp/wd_uss/txt/test2.txt", "remote_src": true, "mode": "0755", "backup": false, "force": true, "is_uss": true, "is_pds": false, "is_src_dir": false, "copy_member": false, "src_member": false, "temp_path": null, "is_mvs_dest": false, "local_charset": "UTF-8", "is_binary": false, "local_follow": true, "ignore_sftp_stderr": false, "validate": false, "unsafe_writes": false, "encoding": null, "content": null, "backup_name": null, "volume": null, "dest_data_set": null, "size": null, "tmp_hlq": null, "owner": null, "group": null, "seuser": null, "serole": null, "selevel": null, "setype": null, "attributes": null}}}\r\n', b'Shared connection to 9.30.222.103 closed.\r\n')
<9.30.222.103> ESTABLISH SSH CONNECTION FOR USER: vcorfma
<9.30.222.103> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="vcorfma"' -o ConnectTimeout=10 -o 'ControlPath="/home/vcorfmat/.ansible/cp/70a06198bc"' 9.30.222.103 '/bin/sh -c '"'"'rm -f -r /u/vcorfma/.ansible/tmp/ansible-tmp-1681474911.0594854-304-278803071325442/ > /dev/null 2>&1 && sleep 0'"'"''
<9.30.222.103> (0, b'', b'')
ok: [zos_host_01] => {
    "changed": false,
    "dest": "/tmp/wd_uss/txt/test2.txt",
    "gid": 100,
    "group": "OMVS",
    "invocation": {
        "module_args": {
            "backup": false,
            "dest": "/tmp/wd_uss/txt/test2.txt",
            "force": true,
            "mode": "0755",
            "remote_src": true,
            "src": "/tmp/input/test2.txt"
        }
    },
    "is_binary": false,
    "mode": "0755",
    "owner": "VCORFMA",
    "size": 23,
    "src": "/SYSTEM/tmp/input/test2.txt",
    "state": "file",
    "uid": 10149
}

### Ansible configuration.

```YAML
#*******************************************************************************
# Licensed Materials - Property of IBM
# (c) Copyright IBM Corp. 2022. All Rights Reserved.
#
# Note to U.S. Government Users Restricted Rights:
# Use, duplication or disclosure restricted by GSA ADP Schedule
# Contract with IBM Corp.
#*******************************************************************************
[defaults]
forks = 25
host_key_checking = False
#callbacks_enabled = timer, profile_tasks, profile_roles
stdout_callback = ibm.ibm_zos_wazi_deploy.cb_evidences

[ssh_connection]
pipelining = True
#ssh_args = -o ControlMaster=auto -o ControlPersist=3600s -o PreferredAuthentications=publickey
```

### Contents of the inventory

```YAML
all:
  hosts:
    cvb1:
      ansible_host: X.X.X.X
      ansible_user: MYUSER
      ansible_ssh_port: 222
```

### Contents of `group_vars` or `host_vars`

```YAML
environment_vars:
  _BPXK_AUTOCVT: "ON"
  ZOAU_HOME: "{{ ZOAU }}"
  #PYTHONPATH: "{{ ZOAU }}/lib"
  LIBPATH: "{{ ZOAU }}/lib:{{ PYZ }}/lib:/lib:/usr/lib:."
  PATH: "{{ ZOAU }}/bin:{{ PYZ }}/bin:/bin:/var/bin"
  _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
  _TAG_REDIR_ERR: "txt"
  _TAG_REDIR_IN: "txt"
  _TAG_REDIR_OUT: "txt"
  LANG: "C"
```

### PR DESCRIPTION
**Pull request by fernandofloresg**
##### SUMMARY

Fixes #737 
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

zos_copy

### CODE REVIEW

**Review by ddimatos — APPROVED** [review]
@fernandofloresg looks good to me, nice catch on adding the second call to `set_mode_if_different(dest, mode, False)`.  If our pipeline did not default to 3.9 we probably would have found this sooner. Thanks for the changes.

File: changelogs/fragments/1064-corruped-second-copy.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+bugfixes:
+  - zos_copy - When performing a copy operation to an existing file, the copied
+    file resulted in having corrupted contents. Fix now implements a workaround
+    to not use the specific copy routine which corrupts the file contents.
```
Comment (AndreMarcel99) [inline changelogs/fragments/1064-corruped-second-copy.yml:4]: Change 'which' for 'that'

**Review by AndreMarcel99 — CHANGES_REQUESTED** [review]

**Review by rexemin — APPROVED** [review]
Looks good

File: changelogs/fragments/1064-corruped-second-copy.yml (hunk 1 above)
Comment (fernandofloresg) [reply changelogs/fragments/1064-corruped-second-copy.yml]: done

**Review by AndreMarcel99 — APPROVED** [review]

**Pull request merged**